src/Repository.php

```diff
@@ -38,14 +57,30 @@ public function register(): void
         }
     }
 
+    /**
+     * @return ComposerFile|array<null>
+     */
     private function getComposerFrom(string $path): array
     {
-        return is_file($path) ? json_decode(file_get_contents($path), true) : [];
+        if (is_file($path)) {
+            /** @var string $content */
+            $content = file_get_contents($path);
+            return json_decode($content, true);
+        }
+        return [];
     }
 
+    /**
+     * @param ComposerFile $composer
+	 * @return class-string<Contracts\IPackage>[]
+	 */
     private function findJalnoPackageFrom(array $composer): array
     {
-        if (!isset($composer["extra"]["jalno"]["package"]) or empty($composer["extra"]["jalno"]["package"])) {
+        if (
+            !isset($composer["extra"]["jalno"]) or
```

Comment:
why you checking `$composer["extra"]["jalno"]`?

Reply:
Cuz `phpstan` warn about that and says the `$composer["extra"]["jalno"]` maybe null.